packages/swing-store/src/snapStore.js

```diff
@@ -223,6 +245,29 @@ export function makeSnapStore(
     );
   }
 
+  const sqlGetSnapshot = db.prepare(`
+    SELECT compressedSnapshot
+    FROM snapshots
+    WHERE vatID = ? AND endPos = ?
+  `);
+  sqlGetSnapshot.pluck(true);
+
+  /**
+   * @param {string} vatID
+   * @param {number} endPos
+   * @returns {AsyncIterable<Uint8Array>}
+   */
+  function getSnapshot(vatID, endPos) {
+    const compressedSnapshot = sqlGetSnapshot.get(vatID, endPos);
+    const gzReader = Readable.from(compressedSnapshot);
+    const unzipper = createGunzip();
+    const snapshotReader = gzReader.pipe(unzipper);
+    async function* fetcher() {
+      yield* snapshotReader;
+    }
+    return fetcher();
+  }
```

Comment:
Why not make the whole thing a generator function? I guess we want to throw synchronously if there is an error during `sqlGetSnapshot.get` ?

If not the following would work just as well:

```suggestion
  async function* getSnapshot(vatID, endPos) {
    const compressedSnapshot = sqlGetSnapshot.get(vatID, endPos);
    const gzReader = Readable.from(compressedSnapshot);
    const unzipper = createGunzip();
    const snapshotReader = gzReader.pipe(unzipper);
    yield* snapshotReader;
  }
```

Reply:
I guess I have another Node.js-vs-ES5 question, does this pattern return a non-Node.js-specific value? Should we be trying to avoid returning Node.js-specific values (like Streams and Buffers, I guess)?

Reply:
Node streams are `AsyncIterable` so `yield *` will simply yield `Buffer` chunks (from the gzip stream). Thus the only node specific part is the runtime instance type of the yielded values. Since it's too costly to copy from a `Buffer` instance into a base `UInt8Array` instance, but the former is simply a derived class of the latter anyway, and we document the return type as the base `Uint8Array`, I argue there is effectively no node specific abstractions leaking through the API.

Reply:
I think there was a set of changes to the code that somehow didn't get pushed.  I'm going to rectify that now and we'll see if some of these comments remain relevant.

Reply:
Hmm, looks like I've got a several incomplete but unrelated changes in flight that make just doing an immediate push problematic.  Anyway, this whole comment thread is irrelevant and I'm just going to mark it as resolved.